default/python/common/configure_logging.py

```diff
@@ -161,7 +161,14 @@ def __init__(self, level):
         }[level]
 
     def emit(self, record):
-        self.logger(str(record.msg) + "\n", str(record.name), str(record.filename),
+        record.message = record.getMessage()
+        msg = str(record.message) + "\n"
+        if record.exc_info:
+            if not isinstance(record.exc_info, tuple):
+                record.exc_info = sys.exc_info()
```

Comment:
Is this safe?
On first sight, this looks like a potential memory leak. The ```sys.exc_info()``` call returns a reference to a stack frame as third tuple element and thus introduces circular references that generally should be resolved by explicit 
deletion.

See https://docs.python.org/2/library/sys.html#sys.exc_info

If you are confident an explicit deletion is not required here, please add some comment here why this is safe.

Reply:
> On first sight, this looks like a potential memory leak. 

from the docs you cite, the issue sounds more like a garbage-collection efficiency issue for us rather than an actual memory leak.  But still worth avoiding if possible, certainly.

> Note Beginning with Python 2.2, such cycles are automatically reclaimed when garbage collection is enabled and they become unreachable, but it remains more efficient to avoid creating cycles.

Reply:
I don't want to argue semantics here. It may technically not be a memory leak but the effect I think is very much the same.

Reference cycles are supposed to not be completely freed if objects defining a ```__del__``` method are involved (seemingly this was resolved in Python 3.4). Instead, it is put into the garbage which must be manually cleared: https://docs.python.org/2/library/gc.html#gc.garbage
To my knowledge we never manually clear the garbage so the memory would never be freed.

It may of course be that there are additionally measures taken in case of tracebacks and it is actually safe to rely on the garbage collector.